query/s3/s3_bucket_policies_restricts_permissions_to_other_aws_account.sql

```diff
@@ -0,0 +1,40 @@
+with crossaccount_buckets as (
+  select
+    distinct arn
+  from
+    aws_s3_bucket,
+    jsonb_array_elements(policy_std -> 'Statement') as s,
+    jsonb_array_elements_text(s -> 'Principal' -> 'AWS') as p,
+    string_to_array(p, ':') as pa,
+    jsonb_array_elements_text(s -> 'Action') as a
+  where
+    s ->> 'Effect' = 'Allow'
+    and (
+      pa [5] != account_id
+      or p = '*'
+    )
+    and a in (
+      's3:deletebucketpolicy',
```

Comment:
Are there other permissions the control cares about? Why were these picked in particular?

Reply:
@cbruno10 The control does care about the other permissions, it is as per the requirement of the control.
https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-standards-fsbp-controls.html#fsbp-s3-6

Reply:
`query/s3/s3_bucket_policy_restricts_cross_account_permission_changes.sql`